### PR TITLE
Make leap_day=True default for PSM3 (deprecation)

### LINKED ISSUE #1481: Set leap_day=True as default in pvlib.iotools.get_psm3()
**Is your feature request related to a problem? Please describe.**
Several functions require equal timesteps with no gaps (e.g., `pvlib.temperature.prilliman` and `detect_clearsky`). Having weather data that excludes leap day will cause an error. Specifically, using `pvlib.iotools.get_psm3` for a leap year with the default `leap_day=True` would be a common reason for encountering this error.

**Describe the solution you'd like**
As suggested by @adriesse here https://github.com/pvlib/pvlib-python/issues/1476#issuecomment-1163130675, perhaps the default should be changed to `leap_day=True`?

**Describe alternatives you've considered**

**Additional context**
Related to https://github.com/pvlib/pvlib-python/issues/1476. 

On the topic of "is this a good idea", some comments from issue 1476:
> "I wonder if there are any compelling reasons to keep it as False, other than the fact that other tools (e.g., NREL SAM) can't handle leap day." 

(by me)

> "I suspect `leap_day=False` is rooted in matching the PSM3 API's decision to set it False by default." 

(by @kanderso-nrel)

### PR DESCRIPTION
**Pull request by AdamRJensen**
- [x] Closes #1481 
 - [x] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing.html)
 - [x] Tests added
 - ~~[ ] Updates entries in [`docs/sphinx/source/reference`](https://github.com/pvlib/pvlib-python/blob/master/docs/sphinx/source/reference) for API changes.~~
 - ~~[ ] Adds description and name entries in the appropriate "what's new" file in [`docs/sphinx/source/whatsnew`](https://github.com/pvlib/pvlib-python/tree/master/docs/sphinx/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).~~
 - [x] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.
 - [x] Pull request is nearly complete and ready for detailed review.
 - [x] Maintainer: Appropriate GitHub Labels (including `remote-data`) and Milestone are assigned to the Pull Request and linked Issue.

It has been decided to change the default value for the `leap_day` argument in the `get_psm3` function from False to True (see #1481). This PR adds a deprecation warning stating that the default will be changed to `leap_day=True` in pvlib 0.11.

### CODE REVIEW

File: pvlib/iotools/psm3.py
Comment (kanderso-nrel) [inline pvlib/iotools/psm3.py:168]: Probably we should not warn if the request is for a TMY/TGY/TDY dataset, right?  No point in warning for cases where the parameter can't have an effect.

**Review by kanderso-nrel — COMMENTED** [review]
Needs a whatsnew entry too

File: pvlib/iotools/psm3.py
Comment (AdamRJensen) [reply pvlib/iotools/psm3.py:168]: Indeed, great point! I have modified it to be the following:

        if (leap_day is None) & (not names.startswith('t')):

File: pvlib/iotools/psm3.py
Comment (kanderso-nrel) [inline pvlib/iotools/psm3.py:168]: Nitpick: `and` and `&` have identical semantics when applied to booleans as far as I know (except short-circuiting), but stylistically I prefer to reserve `&` for situations where it actually behaves differently (e.g. element-wise array logic or bit fiddling).  But I don't think PEP8 opines on this point, and from how often I see people using `&` for scalar booleans I guess a lot of people would disagree.

**Review by kanderso-nrel — APPROVED** [review]
OK with me!

File: pvlib/iotools/psm3.py
Comment (AdamRJensen) [reply pvlib/iotools/psm3.py]: Sure, why not :grinning:

**Comment by AdamRJensen** [comment]
@pvlib/pvlib-maintainer This PR is ready for a second review if anyone wants to take a look

File: docs/sphinx/source/whatsnew/v0.9.2.rst
Comment (cwhanse) [inline docs/sphinx/source/whatsnew/v0.9.2.rst:28]:
```suggestion
* :py:func:`pvlib.iotools.get_psm3` now raises a deprecation warning if
  the `leap_day` parameter is not specified in a single-year request.
```

**Review by cwhanse — APPROVED** [review]

**Pull request merged**
